server/recceiver/test_cfstore.py

```diff
@@ -0,0 +1,234 @@
+                          {u'ch1': abbr(u'ch1', 'testhostc', 3333, 'Inactive'),
+                           u'ch2': abbr(u'ch2', 'testhostc', 3333, 'Inactive'),
+                           u'ch3': abbr(u'ch3', 'testhostb', 2222, 'Inactive'),
+                           u'ch4': abbr(u'ch4', 'testhostc', 3333, 'Inactive'),
+                           u'ch5': abbr(u'ch5', 'testhosta', 1111, 'Inactive'),
+                           u'ch6': abbr(u'ch6', 'testhostc', 3333, 'Inactive'),
+                           u'ch7': abbr(u'ch7', 'testhosta', 1111, 'Inactive')})
+
+        TR9 = mock_TR()
+        TR9.src.host = 'testhostb'
+        TR9.src.port = 2222
+        TR9.addrec = {1: ('ch1', 'longin'), 2: ('ch2', 'longout'), 3: ('ch3', 'stringout'), 7: ('ch7', 'longout')}
+        deferred = yield self.cfstore.commit(TR9)
+        self.assertCommit([u'ch1', u'ch2', u'ch3', u'ch4', u'ch5', u'ch6', u'ch7'],
+                          {u'ch1': abbr(u'ch1', 'testhostb', 2222, 'Active'),
+                           u'ch2': abbr(u'ch2', 'testhostb', 2222, 'Active'),
+                           u'ch3': abbr(u'ch3', 'testhostb', 2222, 'Active'),
+                           u'ch4': abbr(u'ch4', 'testhostc', 3333, 'Inactive'),
+                           u'ch5': abbr(u'ch5', 'testhosta', 1111, 'Inactive'),
+                           u'ch6': abbr(u'ch6', 'testhostc', 3333, 'Inactive'),
+                           u'ch7': abbr(u'ch7', 'testhostb', 2222, 'Active')})
+
+        TR10 = mock_TR()
+        TR10.src.host = 'testhosta'
+        TR10.src.port = 1111
+        TR10.addrec = {1: ('ch1', 'longin'), 2: ('ch2', 'longout'), 3: ('ch3', 'stringout'), 4: ('ch4', 'stringin'), 5: ('ch5', 'longin'), 6: ('ch6', 'stringin'), 7: ('ch7', 'longout')}
+        deferred = yield self.cfstore.commit(TR10)
+        self.assertCommit([u'ch1', u'ch2', u'ch3', u'ch4', u'ch5', u'ch6', u'ch7'],
+                          {u'ch1': abbr(u'ch1', 'testhosta', 1111, 'Active'),
+                           u'ch2': abbr(u'ch2', 'testhosta', 1111, 'Active'),
+                           u'ch3': abbr(u'ch3', 'testhosta', 1111, 'Active'),
+                           u'ch4': abbr(u'ch4', 'testhosta', 1111, 'Active'),
+                           u'ch5': abbr(u'ch5', 'testhosta', 1111, 'Active'),
+                           u'ch6': abbr(u'ch6', 'testhosta', 1111, 'Active'),
+                           u'ch7': abbr(u'ch7', 'testhosta', 1111, 'Active')})
+
+    @defer.inlineCallbacks
+    def test_no_CFS(self):
+        self.cfclient = mock_client()
+        self.cfclient.connected = False
+        self.cfstore.client = self.cfclient
+        self.cfstore.startService()
+        TR1 = mock_TR()
+        TR1.src.host = 'testhosta'
+        TR1.src.port = 1111
+        TR1.addrec = {1: ('ch1', 'longin'), 5: ('ch5', 'longin'), 6: ('ch6', 'stringin'), 7: ('ch7', 'longout')}
+        threading._start_new_thread(self.simulate_reconnect, ())
```

Comment:
Where did you find ```threading._start_new_thread```?  This doesn't appear to be part of the public API for the threading module.  Please use ```threading.Thread```.